src/components/inline_edit/inline_edit_form.test.tsx

```diff
@@ -0,0 +1,303 @@
+      expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
+      expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
+    });
+
+    it('saves text and returns to readMode', () => {
+      const { getByTestSubject, getByText } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+        target: { value: 'New message!' },
+      });
+      expect(
+        getByTestSubject('euiInlineEditModeInput').getAttribute('value')
+      ).toEqual('New message!');
+      fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));
+
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+      expect(getByText('New message!')).toBeTruthy();
+    });
+
+    it('cancels text and returns to readMode', () => {
+      const { getByTestSubject, getByText } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+        target: { value: 'New message!' },
+      });
+      expect(
+        getByTestSubject('euiInlineEditModeInput').getAttribute('value')
+      ).toEqual('New message!');
+      fireEvent.click(getByTestSubject('euiInlineEditModeCancelButton'));
+
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+      expect(getByText('Hello World!')).toBeTruthy();
+    });
+
+    describe('onConfirm behavior on save', () => {
+      it('returns to readMode with updated text when onConfirm returns true', () => {
+        const { getByTestSubject, getByText } = render(
+          <EuiInlineEditForm
+            {...commonInlineEditFormProps}
+            startWithEditOpen={true}
+            onConfirm={() => true}
+          />
+        );
+
+        fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+          target: { value: 'New message!' },
+        });
+        fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));
+
+        expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+        expect(getByText('New message!')).toBeTruthy();
+      });
+
+      it('stays in editMode when onConfirm returns false', () => {
+        const { getByTestSubject, queryByTestSubject } = render(
+          <EuiInlineEditForm
+            {...commonInlineEditFormProps}
+            startWithEditOpen={true}
+            onConfirm={() => false}
+          />
+        );
+
+        fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+          target: { value: 'New message!' },
+        });
+        fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));
+
+        expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
+        expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
+      });
```

Comment:
Now that we've added the `onSave` prop, I'd also like to test that returning onConfirm with false does not trigger onSave

```suggestion
      it('stays in editMode when onConfirm returns false', () => {
        const onSave = jest.fn();

        const { getByTestSubject, queryByTestSubject } = render(
          <EuiInlineEditForm
            {...commonInlineEditFormProps}
            startWithEditOpen={true}
            onSave={onSave}
            onConfirm={() => false}
          />
        );

        fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
          target: { value: 'New message!' },
        });
        fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));

        expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
        expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
      });
      expect(onSave).not.toHaveBeenCalled();
```